server/migrations/201901290642-init.js

```diff
@@ -0,0 +1,164 @@
+                    allowNull: false,
+                },
+            },
+            {},
+        ],
+    },
+    {
+        fn: 'createTable',
+        params: [
+            'Devices',
+            {
+                id: {
+                    type: Sequelize.INTEGER,
+                    primaryKey: true,
+                    autoIncrement: true,
+                },
+                type: {
+                    type: Sequelize.STRING,
+                    allowNull: false,
+                },
+                token: {
+                    type: Sequelize.STRING,
+                    allowNull: false,
+                },
+                isArchived: {
+                    type: Sequelize.BOOLEAN,
+                    defaultValue: false,
+                    allowNull: false,
+                },
+                createdAt: {
+                    type: Sequelize.DATE,
+                    allowNull: false,
+                },
+                updatedAt: {
+                    type: Sequelize.DATE,
+                    allowNull: false,
+                },
+                UserId: {
+                    type: Sequelize.INTEGER,
+                    onUpdate: 'CASCADE',
+                    onDelete: 'SET NULL',
+                    references: {
+                        model: 'Users',
+                        key: 'id',
+                    },
+                    allowNull: true,
+                },
+            },
+            {},
+        ],
+    },
+    {
+        fn: 'createTable',
+        params: [
+            'Notifications',
+            {
+                id: {
+                    type: Sequelize.INTEGER,
+                    primaryKey: true,
+                    autoIncrement: true,
+                },
+                type: {
+                    type: Sequelize.STRING,
+                    allowNull: false,
+                },
+                payload: {
+                    type: Sequelize.JSON,
+                    allowNull: false,
+                },
+                status: {
+                    type: Sequelize.STRING,
+                    allowNull: false,
+                },
+                createdAt: {
+                    type: Sequelize.DATE,
+                    allowNull: false,
+                },
+                updatedAt: {
+                    type: Sequelize.DATE,
+                    allowNull: false,
+                },
+                DeviceId: {
+                    type: Sequelize.INTEGER,
+                    onUpdate: 'CASCADE',
+                    onDelete: 'SET NULL',
+                    references: {
+                        model: 'Devices',
+                        key: 'id',
+                    },
+                    allowNull: true,
+                },
+            },
+            {},
+        ],
+    },
+    {
+        // This manual query is required, rather than using sequelize's addIndex()
+        // because addIndex() does not support IF NOT EXISTS.
+        fn: 'query',
```

Comment:
YMMV: The `describeTable` solution looks a little cleaner than a manual query, see: https://stackoverflow.com/a/49265581/3684641

Reply:
index created automatically alongside unique constraint, see: https://github.com/amida-tech/amida-notification-microservice/pull/37/files#diff-5b4e07fda628098922011c6273e6560dR20

```
-- From psql
\d "Users"
...
Indexes:
    "Users_pkey" PRIMARY KEY, btree (id)
    "Users_uuid_key" UNIQUE CONSTRAINT, btree (uuid)
```